vyper/functions/functions.py

```diff
@@ -1224,6 +1227,61 @@ def _can_compare_with_uint256(operand):
     )
 
 
+@signature('decimal')
+def sqrt(expr, args, kwargs, context):
+    from vyper.functions.utils import (
+        generate_inline_function,
+    )
+    arg = args[0]
+    sqrt_code = """
+assert x >= 0.0
+z: decimal = (x + 1.0) / 2.0
+y: decimal = x
+
+for i in range(256):
+    if (z >= y):
+        break
+    y = z
+    z = (x / z + z) / 2.0
+    """
```

Comment:
@charles-cooper I think this is beginning of our own virtual function implementation :)

Reply:
Neat!